test-tools/wamr-ide/VSCode-Extension/src/utilities/lldbUtilities.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2022 Amazon.com Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+import * as vscode from 'vscode';
+import * as os from 'os';
+import * as path from 'path';
+import * as fs from 'fs';
+import { CheckIfDirectoryExist, DownloadFile, UnzipFile } from './directoryUtilities';
+
+const LLDB_RESOURCE_DIR = "resource/debug";
+const LLDB_OS_DIR_MAP: Partial<Record<NodeJS.Platform, string>> = {
+    "linux": "linux",
+    "darwin": "osx"
+};
+const LLDB_OS_DOWNLOAD_URL_SUFFIX_MAP: Partial<Record<NodeJS.Platform, string>> = {
+    "linux": "x86_64-ubuntu-22.04",
+    "darwin": "universal-macos-latest"
+};
+
+const WamrLLDBNotSupportedError = new Error("WAMR LLDB is not supported on this platform");
+
+function getLLDBUnzipFilePath(destinationFolder: string, filename: string) {
+    const dirs = filename.split("/");
+    if (dirs[0] == "inst") {
+        dirs.shift();
+    }
+
+    return path.join(destinationFolder, ...dirs);
+}
+
+function getLLDBDownloadUrl(context: vscode.ExtensionContext): string {
+    const wamrVersion = require(path.join(context.extensionPath, "package.json")).version;
+    const lldbOsUrlSuffix = LLDB_OS_DOWNLOAD_URL_SUFFIX_MAP[os.platform()];
+
+    if (!lldbOsUrlSuffix) {
+        throw WamrLLDBNotSupportedError;
+    }
+
+    return `https://github.com/bytecodealliance/wasm-micro-runtime/releases/download/WAMR-${wamrVersion}/wamr-lldb-${wamrVersion}-${lldbOsUrlSuffix}.zip`
+}
+
+export function isLLDBInstalled(context: vscode.ExtensionContext): boolean {
+    const extensionPath = context.extensionPath;
+    const lldbOSDir = LLDB_OS_DIR_MAP[os.platform()]
+    if (!lldbOSDir) {
+        throw WamrLLDBNotSupportedError;
+    }
+
+    const lldbBinaryPath = path.join(extensionPath, LLDB_RESOURCE_DIR, lldbOSDir, "bin", "lldb");
+
+    return CheckIfDirectoryExist(lldbBinaryPath);
```

Comment:
Not sure how harmful it is, but I just looked at the function and it actually checks if file exists, but it doesn't check if that file is a directory. Possibly we'd have to fix that function to be more accurate if that's problematic.